### PR TITLE
MULE-6331: Client so_timeout is replaced by responseTimeout

### PR DESCRIPTION
**Pull request by afelisatti**

### CODE REVIEW

File: transports/http/src/test/resources/http-socket-timeout-config.xml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xmlns:http="http://www.mulesoft.org/schema/mule/http"
+      xmlns:test="http://www.mulesoft.org/schema/mule/test"
+      xsi:schemaLocation="
+		http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd
+		http://www.mulesoft.org/schema/mule/test http://www.mulesoft.org/schema/mule/test/current/mule-test.xsd
+		http://www.mulesoft.org/schema/mule/http http://www.mulesoft.org/schema/mule/http/current/mule-http.xsd
+	">
+
+    <configuration defaultResponseTimeout="1000"/>
```
Comment (pablokraan) [inline transports/http/src/test/resources/http-socket-timeout-config.xml]: Would be better if you can use smaller values for timeouts

File: transports/http/src/test/resources/http-socket-timeout-config.xml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    </flow>
+
+    <flow name="noTimeout">
+        <http:inbound-endpoint address="http://localhost:${port}/noTimeout" exchange-pattern="request-response" connector-ref="httpConn"/>
+        <http:outbound-endpoint address="http://localhost:${port}/out" exchange-pattern="request-response" connector-ref="httpConn"/>
+    </flow>
+
+    <flow name="out">
+        <http:inbound-endpoint address="http://localhost:${port}/out" exchange-pattern="request-response" connector-ref="httpConn"/>
+        <test:component waitTime="1500"/>
```
Comment (pablokraan) [inline transports/http/src/test/resources/http-socket-timeout-config.xml]: This is basically a Thread.sleep. Can you avoid it?

Comment (afelisatti) [inline transports/http/src/test/resources/http-socket-timeout-config.xml]: Other than reducing the values, I cannot think of another way of testing the timeout. Any ideas?

File: transports/http/src/test/java/org/mule/transport/http/functional/HttpSocketTimeoutTestCase.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import org.mule.tck.probe.Probe;
+import org.mule.util.concurrent.Latch;
+
+public class HttpSocketTimeoutTestCase extends FunctionalTestCase {
+
+    @Rule
+    public DynamicPort port = new DynamicPort("port");
+
+    private static Latch latch;
+    private static boolean timedout;
```
Comment (pablokraan) [inline transports/http/src/test/java/org/mule/transport/http/functional/HttpSocketTimeoutTestCase.java]: timedout > timedOut or timeouted?

File: transports/http/src/test/java/org/mule/transport/http/functional/HttpSocketTimeoutTestCase.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    private static final int LATCH_TIMEOUT = 300;
```
Comment (pablokraan) [inline transports/http/src/test/java/org/mule/transport/http/functional/HttpSocketTimeoutTestCase.java]: Put constants at the top of the class

File: transports/http/src/test/java/org/mule/transport/http/functional/HttpSocketTimeoutTestCase.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        @Override
+        public MuleEvent process(MuleEvent event) throws MuleException
+        {
+            try
+            {
+                timedout = latch.await(LATCH_TIMEOUT, MILLISECONDS);
+            }
+            catch (InterruptedException e)
+            {
+                //do nothing
```
Comment (pablokraan) [inline transports/http/src/test/java/org/mule/transport/http/functional/HttpSocketTimeoutTestCase.java]: // Do nothing

File: transports/http/src/test/java/org/mule/transport/http/functional/HttpSocketTimeoutTestCase.java (hunk 3 above)
Comment (afelisatti) [inline transports/http/src/test/java/org/mule/transport/http/functional/HttpSocketTimeoutTestCase.java]: timedOut, I'll rename it.

**Pull request merged**